// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: mgmt/v1alpha1/auth.proto

package mgmtv1alpha1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on LoginCliRequest with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *LoginCliRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on LoginCliRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// LoginCliRequestMultiError, or nil if none found.
func (m *LoginCliRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *LoginCliRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Code

	// no validation rules for RedirectUri

	if len(errors) > 0 {
		return LoginCliRequestMultiError(errors)
	}

	return nil
}

// LoginCliRequestMultiError is an error wrapping multiple validation errors
// returned by LoginCliRequest.ValidateAll() if the designated constraints
// aren't met.
type LoginCliRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m LoginCliRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m LoginCliRequestMultiError) AllErrors() []error { return m }

// LoginCliRequestValidationError is the validation error returned by
// LoginCliRequest.Validate if the designated constraints aren't met.
type LoginCliRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e LoginCliRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e LoginCliRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e LoginCliRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e LoginCliRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e LoginCliRequestValidationError) ErrorName() string { return "LoginCliRequestValidationError" }

// Error satisfies the builtin error interface
func (e LoginCliRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sLoginCliRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = LoginCliRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = LoginCliRequestValidationError{}

// Validate checks the field values on LoginCliResponse with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *LoginCliResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on LoginCliResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// LoginCliResponseMultiError, or nil if none found.
func (m *LoginCliResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *LoginCliResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetAccessToken()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, LoginCliResponseValidationError{
					field:  "AccessToken",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, LoginCliResponseValidationError{
					field:  "AccessToken",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetAccessToken()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return LoginCliResponseValidationError{
				field:  "AccessToken",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if len(errors) > 0 {
		return LoginCliResponseMultiError(errors)
	}

	return nil
}

// LoginCliResponseMultiError is an error wrapping multiple validation errors
// returned by LoginCliResponse.ValidateAll() if the designated constraints
// aren't met.
type LoginCliResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m LoginCliResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m LoginCliResponseMultiError) AllErrors() []error { return m }

// LoginCliResponseValidationError is the validation error returned by
// LoginCliResponse.Validate if the designated constraints aren't met.
type LoginCliResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e LoginCliResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e LoginCliResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e LoginCliResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e LoginCliResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e LoginCliResponseValidationError) ErrorName() string { return "LoginCliResponseValidationError" }

// Error satisfies the builtin error interface
func (e LoginCliResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sLoginCliResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = LoginCliResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = LoginCliResponseValidationError{}

// Validate checks the field values on GetAuthStatusRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetAuthStatusRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetAuthStatusRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetAuthStatusRequestMultiError, or nil if none found.
func (m *GetAuthStatusRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *GetAuthStatusRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return GetAuthStatusRequestMultiError(errors)
	}

	return nil
}

// GetAuthStatusRequestMultiError is an error wrapping multiple validation
// errors returned by GetAuthStatusRequest.ValidateAll() if the designated
// constraints aren't met.
type GetAuthStatusRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetAuthStatusRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetAuthStatusRequestMultiError) AllErrors() []error { return m }

// GetAuthStatusRequestValidationError is the validation error returned by
// GetAuthStatusRequest.Validate if the designated constraints aren't met.
type GetAuthStatusRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetAuthStatusRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetAuthStatusRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetAuthStatusRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetAuthStatusRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetAuthStatusRequestValidationError) ErrorName() string {
	return "GetAuthStatusRequestValidationError"
}

// Error satisfies the builtin error interface
func (e GetAuthStatusRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetAuthStatusRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetAuthStatusRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetAuthStatusRequestValidationError{}

// Validate checks the field values on GetAuthStatusResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetAuthStatusResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetAuthStatusResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetAuthStatusResponseMultiError, or nil if none found.
func (m *GetAuthStatusResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *GetAuthStatusResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for IsEnabled

	if len(errors) > 0 {
		return GetAuthStatusResponseMultiError(errors)
	}

	return nil
}

// GetAuthStatusResponseMultiError is an error wrapping multiple validation
// errors returned by GetAuthStatusResponse.ValidateAll() if the designated
// constraints aren't met.
type GetAuthStatusResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetAuthStatusResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetAuthStatusResponseMultiError) AllErrors() []error { return m }

// GetAuthStatusResponseValidationError is the validation error returned by
// GetAuthStatusResponse.Validate if the designated constraints aren't met.
type GetAuthStatusResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetAuthStatusResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetAuthStatusResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetAuthStatusResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetAuthStatusResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetAuthStatusResponseValidationError) ErrorName() string {
	return "GetAuthStatusResponseValidationError"
}

// Error satisfies the builtin error interface
func (e GetAuthStatusResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetAuthStatusResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetAuthStatusResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetAuthStatusResponseValidationError{}

// Validate checks the field values on AccessToken with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *AccessToken) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on AccessToken with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in AccessTokenMultiError, or
// nil if none found.
func (m *AccessToken) ValidateAll() error {
	return m.validate(true)
}

func (m *AccessToken) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for AccessToken

	// no validation rules for ExpiresIn

	// no validation rules for Scope

	// no validation rules for TokenType

	if m.RefreshToken != nil {
		// no validation rules for RefreshToken
	}

	if m.IdToken != nil {
		// no validation rules for IdToken
	}

	if len(errors) > 0 {
		return AccessTokenMultiError(errors)
	}

	return nil
}

// AccessTokenMultiError is an error wrapping multiple validation errors
// returned by AccessToken.ValidateAll() if the designated constraints aren't met.
type AccessTokenMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m AccessTokenMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m AccessTokenMultiError) AllErrors() []error { return m }

// AccessTokenValidationError is the validation error returned by
// AccessToken.Validate if the designated constraints aren't met.
type AccessTokenValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e AccessTokenValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e AccessTokenValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e AccessTokenValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e AccessTokenValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e AccessTokenValidationError) ErrorName() string { return "AccessTokenValidationError" }

// Error satisfies the builtin error interface
func (e AccessTokenValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sAccessToken.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = AccessTokenValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = AccessTokenValidationError{}

// Validate checks the field values on GetAuthorizeUrlRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetAuthorizeUrlRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetAuthorizeUrlRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetAuthorizeUrlRequestMultiError, or nil if none found.
func (m *GetAuthorizeUrlRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *GetAuthorizeUrlRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for State

	// no validation rules for RedirectUri

	// no validation rules for Scope

	if len(errors) > 0 {
		return GetAuthorizeUrlRequestMultiError(errors)
	}

	return nil
}

// GetAuthorizeUrlRequestMultiError is an error wrapping multiple validation
// errors returned by GetAuthorizeUrlRequest.ValidateAll() if the designated
// constraints aren't met.
type GetAuthorizeUrlRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetAuthorizeUrlRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetAuthorizeUrlRequestMultiError) AllErrors() []error { return m }

// GetAuthorizeUrlRequestValidationError is the validation error returned by
// GetAuthorizeUrlRequest.Validate if the designated constraints aren't met.
type GetAuthorizeUrlRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetAuthorizeUrlRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetAuthorizeUrlRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetAuthorizeUrlRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetAuthorizeUrlRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetAuthorizeUrlRequestValidationError) ErrorName() string {
	return "GetAuthorizeUrlRequestValidationError"
}

// Error satisfies the builtin error interface
func (e GetAuthorizeUrlRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetAuthorizeUrlRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetAuthorizeUrlRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetAuthorizeUrlRequestValidationError{}

// Validate checks the field values on GetAuthorizeUrlResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetAuthorizeUrlResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetAuthorizeUrlResponse with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetAuthorizeUrlResponseMultiError, or nil if none found.
func (m *GetAuthorizeUrlResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *GetAuthorizeUrlResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Url

	if len(errors) > 0 {
		return GetAuthorizeUrlResponseMultiError(errors)
	}

	return nil
}

// GetAuthorizeUrlResponseMultiError is an error wrapping multiple validation
// errors returned by GetAuthorizeUrlResponse.ValidateAll() if the designated
// constraints aren't met.
type GetAuthorizeUrlResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetAuthorizeUrlResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetAuthorizeUrlResponseMultiError) AllErrors() []error { return m }

// GetAuthorizeUrlResponseValidationError is the validation error returned by
// GetAuthorizeUrlResponse.Validate if the designated constraints aren't met.
type GetAuthorizeUrlResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetAuthorizeUrlResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetAuthorizeUrlResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetAuthorizeUrlResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetAuthorizeUrlResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetAuthorizeUrlResponseValidationError) ErrorName() string {
	return "GetAuthorizeUrlResponseValidationError"
}

// Error satisfies the builtin error interface
func (e GetAuthorizeUrlResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetAuthorizeUrlResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetAuthorizeUrlResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetAuthorizeUrlResponseValidationError{}

// Validate checks the field values on GetCliIssuerRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetCliIssuerRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetCliIssuerRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetCliIssuerRequestMultiError, or nil if none found.
func (m *GetCliIssuerRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *GetCliIssuerRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return GetCliIssuerRequestMultiError(errors)
	}

	return nil
}

// GetCliIssuerRequestMultiError is an error wrapping multiple validation
// errors returned by GetCliIssuerRequest.ValidateAll() if the designated
// constraints aren't met.
type GetCliIssuerRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetCliIssuerRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetCliIssuerRequestMultiError) AllErrors() []error { return m }

// GetCliIssuerRequestValidationError is the validation error returned by
// GetCliIssuerRequest.Validate if the designated constraints aren't met.
type GetCliIssuerRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetCliIssuerRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetCliIssuerRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetCliIssuerRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetCliIssuerRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetCliIssuerRequestValidationError) ErrorName() string {
	return "GetCliIssuerRequestValidationError"
}

// Error satisfies the builtin error interface
func (e GetCliIssuerRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetCliIssuerRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetCliIssuerRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetCliIssuerRequestValidationError{}

// Validate checks the field values on GetCliIssuerResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetCliIssuerResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetCliIssuerResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetCliIssuerResponseMultiError, or nil if none found.
func (m *GetCliIssuerResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *GetCliIssuerResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for IssuerUrl

	// no validation rules for Audience

	if len(errors) > 0 {
		return GetCliIssuerResponseMultiError(errors)
	}

	return nil
}

// GetCliIssuerResponseMultiError is an error wrapping multiple validation
// errors returned by GetCliIssuerResponse.ValidateAll() if the designated
// constraints aren't met.
type GetCliIssuerResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetCliIssuerResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetCliIssuerResponseMultiError) AllErrors() []error { return m }

// GetCliIssuerResponseValidationError is the validation error returned by
// GetCliIssuerResponse.Validate if the designated constraints aren't met.
type GetCliIssuerResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetCliIssuerResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetCliIssuerResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetCliIssuerResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetCliIssuerResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetCliIssuerResponseValidationError) ErrorName() string {
	return "GetCliIssuerResponseValidationError"
}

// Error satisfies the builtin error interface
func (e GetCliIssuerResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetCliIssuerResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetCliIssuerResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetCliIssuerResponseValidationError{}

// Validate checks the field values on AuthUser with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *AuthUser) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on AuthUser with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in AuthUserMultiError, or nil
// if none found.
func (m *AuthUser) ValidateAll() error {
	return m.validate(true)
}

func (m *AuthUser) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for Name

	// no validation rules for Email

	// no validation rules for Image

	if len(errors) > 0 {
		return AuthUserMultiError(errors)
	}

	return nil
}

// AuthUserMultiError is an error wrapping multiple validation errors returned
// by AuthUser.ValidateAll() if the designated constraints aren't met.
type AuthUserMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m AuthUserMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m AuthUserMultiError) AllErrors() []error { return m }

// AuthUserValidationError is the validation error returned by
// AuthUser.Validate if the designated constraints aren't met.
type AuthUserValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e AuthUserValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e AuthUserValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e AuthUserValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e AuthUserValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e AuthUserValidationError) ErrorName() string { return "AuthUserValidationError" }

// Error satisfies the builtin error interface
func (e AuthUserValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sAuthUser.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = AuthUserValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = AuthUserValidationError{}

// Validate checks the field values on GetAuthUserRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetAuthUserRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetAuthUserRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetAuthUserRequestMultiError, or nil if none found.
func (m *GetAuthUserRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *GetAuthUserRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for UserId

	// no validation rules for AuthProvider

	if len(errors) > 0 {
		return GetAuthUserRequestMultiError(errors)
	}

	return nil
}

// GetAuthUserRequestMultiError is an error wrapping multiple validation errors
// returned by GetAuthUserRequest.ValidateAll() if the designated constraints
// aren't met.
type GetAuthUserRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetAuthUserRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetAuthUserRequestMultiError) AllErrors() []error { return m }

// GetAuthUserRequestValidationError is the validation error returned by
// GetAuthUserRequest.Validate if the designated constraints aren't met.
type GetAuthUserRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetAuthUserRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetAuthUserRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetAuthUserRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetAuthUserRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetAuthUserRequestValidationError) ErrorName() string {
	return "GetAuthUserRequestValidationError"
}

// Error satisfies the builtin error interface
func (e GetAuthUserRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetAuthUserRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetAuthUserRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetAuthUserRequestValidationError{}

// Validate checks the field values on GetAuthUserResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetAuthUserResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetAuthUserResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetAuthUserResponseMultiError, or nil if none found.
func (m *GetAuthUserResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *GetAuthUserResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetUser()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, GetAuthUserResponseValidationError{
					field:  "User",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, GetAuthUserResponseValidationError{
					field:  "User",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetUser()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return GetAuthUserResponseValidationError{
				field:  "User",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if len(errors) > 0 {
		return GetAuthUserResponseMultiError(errors)
	}

	return nil
}

// GetAuthUserResponseMultiError is an error wrapping multiple validation
// errors returned by GetAuthUserResponse.ValidateAll() if the designated
// constraints aren't met.
type GetAuthUserResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetAuthUserResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetAuthUserResponseMultiError) AllErrors() []error { return m }

// GetAuthUserResponseValidationError is the validation error returned by
// GetAuthUserResponse.Validate if the designated constraints aren't met.
type GetAuthUserResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetAuthUserResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetAuthUserResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetAuthUserResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetAuthUserResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetAuthUserResponseValidationError) ErrorName() string {
	return "GetAuthUserResponseValidationError"
}

// Error satisfies the builtin error interface
func (e GetAuthUserResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetAuthUserResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetAuthUserResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetAuthUserResponseValidationError{}
